TicTacToeProject/MinimaxStrategy.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace TicTacToeApp
+{
+    public class MinimaxStrategy : IStrategy
+    {
+        public int bestMove;
+        public Board originalBoard;
+        public Rules rules;
+        public string currentPlayerMarker;
+        public string opponentPlayerMarker;
+
+        public MinimaxStrategy(Board originalBoard)
+        {
+            this.rules = new Rules();            
+            this.originalBoard = originalBoard;    
+        }
+
+        public void setPlayers()
+        {
+            var current = originalBoard.getCurrentPlayer();
+            if (current == Symbols.P1_MARKER)
+            {
+                currentPlayerMarker = current;
+                opponentPlayerMarker = Symbols.P2_MARKER;
+            }
+            else
+            {
+                currentPlayerMarker = Symbols.P2_MARKER;
+                opponentPlayerMarker = Symbols.P1_MARKER;
+            }
+        }
+
+        public string getMove()
+        {
+            setPlayers();            
+            var depth = 0;
+            var score = minimax(originalBoard, currentPlayerMarker, opponentPlayerMarker, depth);
```

Comment:
The call to `setPlayers()` and then using `currentPlayerMarker` and `opponentPlayerMarker` is a bit confusing. It's not clear that there's a coupling there. For example, you can't depend on currentPlayerMarker/oppenentPlayerMarker until you've called `setPlayers` -- that's called a temporal coupling. Something *has* to happen before the other thing. That's usually difficult to spot in code. 

The method signature here `minimax(originalBoard, currentPlayerMarker, opponentPlayerMarker, depth)` is 👍 -- I would argue that should be how you execute the strategy, instead of instantiating it with everything it needs. Just pass those things to `getMove(board, currentMarker, opposingMarker, depth = 0)`

We can talk about this more tomorrow if it's helpful.